sourcetracker/_plot.py

```diff
@@ -10,16 +10,306 @@
+                add_line = tra + "_pairedheatmap_nounknowns.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.out_name + add_line))
+
+    def ST_Stacked_bar(self, unknowns=True, x_lab="Sink",
+                       y_lab="Source Proportion", coloring=[], flipped=False):
+        """
+        Creates a Stacked bar plot for the user with direct png save function
+        Parameters
+        ----------
+        unknowns bool
+            removes unknown column
+        xlabel string
+            x label
+        ylabel string
+            y label
+        coloring string list
+            string list of colors to encode the bar plot.
+            must have and equal number of colors as sources
+        flipped bool
+            flips x and y axis
+        returns
+        --------
+        none
+        outputs a stacked bar visualization in a PNG
+
+        color example list
+
+        '#1f77b4'Blue, '#ff7f0e'Orange, '#2ca02c'Green, '#d62728'Red,
+        '#9467bd'Purple, '#8c564b'Brown, '#e377c2'Pink, '#7f7f7f'Grey,
+        '#bcbd22'Gold, '#17becf'Cyan
+        make sure to use contrasting colors in order better illuminate
+        your data above are some example codes to use
+        """
+        prop = self.mpm
+        if not unknowns:
+            prop = prop.drop(['Unknown'], axis=1)
+            prop = prop.div(prop.sum(axis=1), axis=0)
+        if flipped:
+            prop = prop.T
+            y_lab_flip = x_lab
+            x_lab_flip = y_lab
+            y_lab = y_lab_flip
+            x_lab = x_lab_flip
+            prop = prop.div(prop.sum(axis=1), axis=0)
+        prop = prop.reset_index()
+        if len(coloring) != (prop.shape[1]-1):
+            coloring = []
+        if coloring != []:
+            prop.plot(kind='bar', x=prop.columns[0], stacked=True,
+                      figsize=((prop.shape[1] * 3 / 4)+4,
+                      (prop.shape[0] * 3 / 4)+4),
+                      color=coloring)
+        else:
+            prop.plot(kind='bar', x=prop.columns[0], stacked=True,
+                      figsize=((prop.shape[1] * 3 / 4)+4,
+                      (prop.shape[0] * 3 / 4)+4))
+        plt.xlabel(x_lab)
+        plt.ylabel(y_lab)
+        plt.title(self.title)
+        plt.autoscale()
+        plt.xticks(rotation=45, ha='right')
+        if unknowns and flipped:
+            plt.savefig(os.path.join(self.file,
+                                     self.out_name + "_flip_stacked_bar.png"))
+        elif unknowns:
+            plt.savefig(os.path.join(self.file,
+                                     self.out_name + "_stacked_bar.png"))
+        else:
+            add_line = "_stacked_bar_nounknowns.png"
+            if flipped:
+                add_line = "flipped_stacked_bar_nounknowns.png"
+            plt.savefig(os.path.join(self.file,
+                                     self.out_name + add_line))
+
+    def ST_bar(self, unknowns=True, x_lab="Sink",
```

Comment:
What is the benefit of bar plot? 

it seems like the only difference in this code is that stacked=false. Is it possible to merge these methods and just set stacked = bool based on user input?

Reply:
If you would like to do that I would recommend removing ST_bar entirely. I have not found an application that would need a regular bar plot that would not be better served by using the stacked bar. It is a vestigial structure of code that I would rather remove all together than leave a variable to trigger the regular bar plot. I just do not think it is worth keeping. Let me know which you would rather I do.

Reply:
I am not sure that this is vestigial structure of code as it hasn't existed previously in this code base. 

I can't think of a way to reasonable use ST_bar.  If you think its best to remove, instead of refactoring, that is fine with me.

Reply:
Sounds perfect. To be clear it is vestigial in my code as I experimented with ST_bar and came up empty.